lib/internal/timers.js

```diff
@@ -559,6 +576,12 @@ function getTimerCallbacks(runNextTicks) {
         continue;
       }
 
+      let priorContextFrame;
+      if (hasAsyncContextFrame) {
+        priorContextFrame =
+          AsyncContextFrame.exchange(timer[async_context_frame]);
+      }
```

Comment:
It should be supported by AsyncResource so I _think_ that additional stuff there may not be necessary anymore, but I will need to check it again. Agreed that we would not want to have yet another API to do this.

Though long-term my aim is to migrate the ecosystem to AsyncContext.bind rather than using AsyncResource whenever that proposal gets an implementation. AsyncResource is an implementation detail that shouldn't be surfaced. It's less bad than the level of internals/implementation details exposed by async_hooks, but still not nearly as clean as it should be.

The main reason I went for direct use in some places in core was that I was originally trying to see if it would make sense to have a lower-level API to work with in core where we can be more responsible but only allow it to be triggered externally via AsyncResource. That would theoretically give us a little more power to write a bit better performing core code, so long as we're careful about how we use it. Don't need additional closures this way.

Reply:
Ah, right...looking closer, timers don't actually use `AsyncResource`, they build their own custom `AsyncResource`-like thing, thus the need to manually handle this here. I think this is fine as-is. External users would still just get this support through the changes to `AsyncResource`, so nothing to do here.